### PR TITLE
Web UI: Rework the Attach Device section to be universal

### PR DESCRIPTION
**Pull request by rdmark**
What was known as the "Attach Peripheral Device" has been reworked as "Attach Device" and now lists all devices that the piscsi backend can support. As part of this change, Attach Device has been reimagined as the primary way one would attach a disk image as well, and moved up above the Image Management section.

- Add "image file" dropdown for each "supports_file" device type under Attach Device
- Under the hood, merge the two attach endpoints to reduce code duplication
- Move Attach Device further up on the page
- Collapse the list of images by default
- Remove the logic that checked for overflowing bytes in images (complexity for little payoff)
- Don't abort the Web UI when working dirs don't exist, but rather hide sections and show a warning message
- Correct the German translation for Key
- Refactor the integration tests to use globally defined endpoint constants

### CODE REVIEW

**Comment by uweseimet** [comment]
@rdmark Yes, the SAHD type is offered now. What confuses me is that an image file is displayed for "SCHD", but not for any other drive. In the "Attach Device" section I would expect no image files to be attached by default.
Screenshot_20231201_173323
Instead of "Attach Device" the German UI should say something like "Gerät hinzufügen". And instead of "Image-Dateiverwaltung" it should say "Imagedatei-Verwaltung", or better "Image-Datei-Verwaltung". Strictly speaking it should also say "Datei-Upload-Funktionalität" instead of "Dateiupload-Funtionalität".

**Comment by rdmark** [comment]
@uweseimet The SCHD device cannot be attached without specifying an image file, can it? That’s why I filtered out the “None” option for it. If SAHD was a known device type I’d do the same for it.  

Thanks for the German localization suggestions. I’ll get them pushed to the branch later.

**Comment by uweseimet** [comment]
@rdmark Yes, you have to provide an image file for SCHD, but I'm not sure whether the UI should suggest one. From all the image files I have in my folder, it suggests most likely one I am not interested in. IMO some kind of a placeholder would be better, if technically possible. The UI might say in the color red "Select an image file", for instance.
Does the UI select a different filename if the one it would have suggested (aranym.img in my case) is already attached? How does the UI decide which name to suggest?

**Comment by rdmark** [comment]
@uweseimet I like the idea of having the default selected item being something like "Choose an image file". This change has been pushed now, alongside a number of small improvements. For the removable drive types, I still want to allow attaching them without specifying an image file. This is a handy workflow when when you want prepare for inserting an image after booting the host system, or preparing a set of devices to be saved as a default configuration.

The logic for creating the list of images to suggest for a particular device type is: 
- Inquiry piscsi for a list of known supported file endings (`server_info.mapping_info.mapping`) for the device types
- Filter images present in the images dir that match these file endings, for the particular device type, e.g. SCHD
- Sort the resulting list of files in ascending alphabetical order (non-case sensitive)

So basically, the first matching file alphabetically gets suggested by default.

**Comment by uweseimet** [comment]
@rdmark I noticed that for SCRM you also ask the user for choosing an image file. Maybe this should better say "Optionally choose an image file" or something similar? Otherwise it's not clear for the user that you can simply proceed without attaching a file yet.

**Comment by rdmark** [comment]
@uweseimet You seem to not be running the very latest code there. You may have to force the web UI to restart.

I agree that the alignment is a bit messy right now. Let me think about how to make it cleaner. However there is only so much you can with the nesting of web forms and tables, while also making it work on very old browsers running on vintage computers.

Regarding translations, our process right now is to ask for community members to translate 2 weeks before we plan to tag a release. Web UI strings are often in flux as we're adding and refining features.

**Comment by rdmark** [comment]
@uweseimet If you expand the dropdowns you should be able to see that the "Choose a file" option is greyed out. (It's up to the browser how it wants to render the `disabled` property.)

> How do you know that it is a fixed disk drive? It could be anything. In case you derive this from "HD" at the end: The UI should not try to be too smart. This can backfire ;-).

I know by asking piscsi if the device is `supports_file` but not `removable`.

**Comment by rdmark** [comment]
I pushed a small css style update that adds a width style to the attach device dropdowns, so that they line up under most circumstances. Tested that it looks good on my iPhone as well.

**Comment by uweseimet** [comment]
> @uweseimet If you expand the dropdowns you should be able to see that the "Choose a file" option is greyed out. (It's up to the browser how it wants to render the `disabled` property.)

Are you sure? Neither Firefox nor Chrome, the latter having the biggest market share by far, grey it out on my machine.

> > How do you know that it is a fixed disk drive? It could be anything. In case you derive this from "HD" at the end: The UI should not try to be too smart. This can backfire ;-).
> 
> I know by asking piscsi if the device is `supports_file` but not `removable`.

Looks as if we both agree that #480 will never be implemented. Otherwise the UI would say "Fixed Disk Drive" when there is a tape drive. That's what I meant with being too smart ;-).

**Review by uweseimet — APPROVED** [review]

**Comment by rdmark** [comment]
Screenshot 2023-12-08 at 10 23 36

I fed some mock data via piscsi_cmds.get_device_types() to test the happy path:

```

        device_types["SAHD"] = {'removable': False, 'supports_file': True, 'params': {}, 'block_sizes': [4096, 2048, 1024, 512]}
        device_types["SARM"] = {'removable': True, 'supports_file': True, 'params': {}, 'block_sizes': [4096, 2048, 1024, 512]}
        device_types["SAAA"] = {'removable': False, 'supports_file': False, 'params': {'foo': '123', 'bar': 'abc'}}
```
